dms/static/src/js/fields/preview.js

```diff
@@ -28,7 +28,7 @@ odoo.define("dms.preview", function(require) {
                 "/" +
                 this.fieldName +
                 "/" +
-                "datas" +
+                this.activeAttachment.name +
```

Comment:
Are you sure about this?

Reply:
Yes, I am using https://github.com/odoo/odoo/blob/12.0/addons/web/controllers/main.py#L1029

Reply:
But is that route trustable? May there be conflicts with the name?

Reply:
Isn't better to use the route without that part?

Reply:
With your proposal the downloaded file will be named as datas.pdf or datas.jpg. 
With mine, the name of the file, will be the name of the file on odoo.

Not a big issue for me, I can change it if you think is better :wink:

Reply:
Ah, OK. Is this handling correctly accented chars and so on?

Reply:
I tried several options and I found one that could give problems, so I am checking again :wink:

Reply:
OK, let me know :smiley:

Reply:
Maybe a call to `slug` over that name (is available at this level? or you need a computed field for that?)

Reply:
Well, I used the URLSearchParams in order to remove the problem, so, now it should work